proxy/coco/src/peer/run_state.rs

```diff
@@ -411,12 +411,16 @@ impl RunState {
     /// Handle [`ControlInput`]s.
     fn handle_control(&mut self, input: ControlInput) -> Vec<Command> {
         match input {
-            ControlInput::CancelRequest(urn, timestamp, sender) => vec![Command::Control(
-                ControlCommand::Respond(control::Response::CancelSearch(
-                    sender,
-                    self.waiting_room.canceled(&urn, timestamp),
-                )),
-            )],
+            ControlInput::CancelRequest(urn, timestamp, sender) => {
+                let request = self.waiting_room.canceled(&urn, timestamp).map(|()| {
+                    self.waiting_room
+                        .remove(&urn)
+                        .expect("URN was found for cancel")
```

Comment:
Can we avoid panics in the `RunState`. It's a sign that we didn't hold it right.

Reply:
Also why would we panic on a `None`, it's fine to make the `Option` part of the response, then the caller can decide if it was unexpected behaviour.

Reply:
Well it shouldn't _be_ `None` because we're in the `Ok` case of the call to `cancelled`. Happy to return the `Option` regardless.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1238/commits/55e50aef3eb906e4026afd93e7894030164ef451